.github/workflows/unit.yml

```diff
@@ -77,6 +77,10 @@ jobs:
         flags: unit
         file: ./unit-coverage.lcov
 
+    # Runs here because it needs the roundtrip proto.
+    - name: yarn test-viewer
+      run: yarn build-viewer && xvfb-run --auto-servernum bash $GITHUB_WORKSPACE/.github/scripts/test-retry.sh yarn test-viewer
```

Comment:
nit: could we do the `yarn build-viewer` step higher up with other setup stuff. Like after `yarn build-report`